File: typescript/packages/bench/src/notifiers/discordNotifier.ts
/** Discord webhook transport. */

import type { Delivery } from "../contracts/notifierContract";
import { AbstractNotifier } from "../support/abstractNotifier";

/** Discord webhook transport. */
export class DiscordNotifier extends AbstractNotifier {
    /** Channel name reported on every delivery receipt. */
    static readonly CHANNEL = "discord";

    constructor(endpoint = "https://discord.test/hook") {
        super(DiscordNotifier.CHANNEL, 2000, endpoint);
    }

    override send(subject: string, body: string): Delivery {
        this.guard(subject, body);

        return { channel: this.channel, reference: `${this.channel}:${subject}` };
    }
}

File: python/atelier_bench/notifiers/discord_notifier.py
"""discord notifier."""

from __future__ import annotations

from ..contracts.notifier_contract import Delivery
from ..support.abstract_notifier import AbstractNotifier


class DiscordNotifier(AbstractNotifier):
    CHANNEL = "discord"

    def __init__(self, endpoint: str = "https://discord.test/hook") -> None:
        super().__init__(self.CHANNEL, 2000, endpoint)

    def send(self, subject: str, body: str) -> Delivery:
        self.guard(subject, body)
        return self.deliver(subject)
